### PR TITLE
New versions: py-gpaw, py-ase

### PR DESCRIPTION
**Pull request by prckent**
Also handle updated configuration file name in newer versions.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-ase/package.py [hunk 1]
```diff
@@ -14,13 +14,22 @@ class PyAse(PythonPackage):
...
+    version('3.18.0', sha256='39d45f12def2669605bffc82926acfb13a0d0610e6d82740fa316aafa70f97f9')
     version('3.15.0', sha256='5e22d961b1311ef4ba2d83527f7cc7448abac8cf9bddd1593bee548459263fe8')
     version('3.13.0', sha256='c4046c50debac28415b36616d79aa28e68ae2cd03c013c2aed6a1e3d465c0ee1')
 
     depends_on('python@2.6:', type=('build', 'run'), when='@:3.15.0')
     depends_on('python@3.5:', type=('build', 'run'), when='@3.18.0:')
+    depends_on('python@3.6:', type=('build', 'run'), when='@3.20.0:')
     depends_on('py-numpy', type=('build', 'run'))
     depends_on('py-matplotlib', type=('build', 'run'))
     depends_on('py-scipy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ase/package.py]: Flask is no longer required. The rest have specific version requirements now:
```python
install_requires = [                                                                     
    'numpy>=1.11.3',                                                                     
    'scipy>=0.18.1',                                                                     
    'matplotlib>=2.0.0',                                                                 
]   
```

File: var/spack/repos/builtin/packages/py-gpaw/package.py [hunk 2]
```diff
@@ -26,11 +29,15 @@ class PyGpaw(PythonPackage):
     depends_on('mpi', when='+mpi', type=('build', 'link', 'run'))
     depends_on('python@2.6:', type=('build', 'run'), when='@:1.3.0')
     depends_on('python@3.5:', type=('build', 'run'), when='@19.8.1:')
+    depends_on('python@3.6:', type=('build', 'run'), when='@20.10.0:')
     depends_on('py-ase@3.13.0:', type=('build', 'run'), when='@1.3.0')
     depends_on('py-ase@3.18.0:', type=('build', 'run'), when='@19.8.1')
+    depends_on('py-ase@3.19.0:', type=('build', 'run'), when='@20.1.0')
+    depends_on('py-ase@3.20.1:', type=('build', 'run'), when='@20.10.0')
+    depends_on('py-ase@3.21.0:', type=('build', 'run'), when='@21.1.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gpaw/package.py:37]: gpaw 21.1.0 requires ase 3.20.1:

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-ase/package.py (hunk 1 above)
Comment (prckent) [reply var/spack/repos/builtin/packages/py-ase/package.py]: Updated

File: var/spack/repos/builtin/packages/py-gpaw/package.py (hunk 2 above)
Comment (prckent) [reply var/spack/repos/builtin/packages/py-gpaw/package.py:37]: No. ase 3.21.0 is required per the official release notes https://wiki.fysik.dtu.dk/gpaw/releasenotes.html#version-21-1-0

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-gpaw/package.py:37]: Hmm, I tend to trust the `setup.py` more than the documentation, but I suppose they might have forgotten to update it...

File: var/spack/repos/builtin/packages/py-ase/package.py [hunk 3]
```diff
@@ -14,15 +14,23 @@ class PyAse(PythonPackage):
...
+    version('3.18.0', sha256='39d45f12def2669605bffc82926acfb13a0d0610e6d82740fa316aafa70f97f9')
     version('3.15.0', sha256='5e22d961b1311ef4ba2d83527f7cc7448abac8cf9bddd1593bee548459263fe8')
     version('3.13.0', sha256='c4046c50debac28415b36616d79aa28e68ae2cd03c013c2aed6a1e3d465c0ee1')
 
     depends_on('python@2.6:', type=('build', 'run'), when='@:3.15.0')
     depends_on('python@3.5:', type=('build', 'run'), when='@3.18.0:')
-    depends_on('py-numpy', type=('build', 'run'))
-    depends_on('py-matplotlib', type=('build', 'run'))
-    depends_on('py-scipy', type=('build', 'run'))
-    depends_on('py-flask', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ase/package.py:27]: I assume older versions still need flask...

Comment (prckent) [reply var/spack/repos/builtin/packages/py-ase/package.py:27]: Fixed

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**